diagnosis/collectlogs.sh

```diff
@@ -203,13 +244,18 @@ if systemctl list-units | grep -q etcd.service; then
 fi
 
 if systemctl list-units | grep -q docker.service; then
-    collectMobyMetadata
+    collectDockerMetadata
     sudo journalctl -n 10000 --utc -o short-iso -u docker &>> ${LOGDIRECTORY}/daemons/k8s-docker.log
 fi
 
+if systemctl list-units | grep -q containerd.service; then
+    collectContainerdMetadata
+    sudo journalctl -n 10000 --utc -o short-iso -u containerd &>> ${LOGDIRECTORY}/daemons/k8s-containerd.log
```

Comment:
When using`journalctl`, I would use flag `-r` (reverse), otherwise there is a chance you may miss the last entries. A combination of both `-n` and `-r` would be ideal

```
timeout 15 journalctl --utc -o short-iso --no-pager -n XXX -u ${1} &> the.log
timeout 15 journalctl --utc -o short-iso --no-pager -r -u ${1} &> the.end
tac the.end >> the.log
```

Reply:
checkout this for more ideas
https://github.com/Azure/aks-engine/blob/master/scripts/collect-logs.sh